### PR TITLE
Initial HTTP Server

### PR DESCRIPTION
**Pull request by arb**
Closes #1

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "hueniversity",
+  "version": "1.0.0",
```
Comment (calmdev) [inline package.json]: The assignment calls for a version of `0.0.1` in the payload

File: index.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+// Load Modules
```
Comment (petermilan) [inline index.js]: I think that this file should contain:
```
module.exports = require('./lib');
```
(the same approach is used inside of hapi)

File: index.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        handler: function (request, reply) {
+
+            reply({
+                version: internals.package.version
+            });
+        }
+    }
+});
+
+server.start(function () {
```
Comment (petermilan) [inline index.js]: Could we somehow handle the error? The callback should have an err argument:

```

File: index.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
+// Load Modules
+var Hapi = require('hapi');
+
+//Declare internals
+var internals = {
+    package: require('./package.json')
+};
+
+var server = new Hapi.Server();
```
Comment (calmdev) [inline index.js]: Can we use the debug option with `good console`? 

```js
var server = new Hapi.Server({
	debug: {
		request: ['error', 'uncaught']
	}
});
```

File: index.js (hunk 3 above)
Comment (calmdev) [inline index.js]: What about using `server.log`?

```js
server.log(['error'], 'Failed to start server: ' + error);
server.log(['info'], 'Server running at: ' + server.info.uri);
```

File: index.js (hunk 4 above)
Comment (hueniverse) [inline index.js]: We're not there yet... no external plugins for now.

File: index.js (hunk 3 above)
Comment (petermilan) [inline index.js]: looks good for me

File: index.js (hunk 4 above)
Comment (calmdev) [inline index.js]: Ok

File: index.js (hunk 3 above)
Comment (arb) [inline index.js]: The problem with using `server.log` is that for this simple example, there isn't anything listening to the events this ultimately emits. Logging to the console is simpler for this first pass.

File: index.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
+// Load Modules
+var Hapi = require('hapi');
+
+//Declare internals
+var internals = {
```
Comment (fyockm) [inline index.js]: Why declare a separate `internals` object? Verses simply:
```
var package: require()
```

Comment (AdriVanHoudt) [inline index.js]: see bottom http://hapijs.com/styleguide

Comment (AdriVanHoudt) [inline index.js]: actual explanation https://gist.github.com/hueniverse/a06f6315ea736ed1b46d#module-classes

File: index.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+server.route({
+    method: 'GET',
+    path: '/version',
+    config: {
+        handler: function (request, reply) {
+
+            reply({
+                version: internals.package.version
+            });
```
Comment (hueniverse) [inline index.js]: `return reply()` reads better. Also more efficient to prepare the full response object in advance. In this case it won't make any different but if the static response object was bigger, there is no reason to create and destroy the exact same object over and over again.

File: index.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+        }
+    }
+});
+
+server.start(function (err) {
+
+    if (err) {
+        console.error('There was an error starting the server: ' + err);
+        process.exit(1);
+    }
```
Comment (hueniverse) [inline index.js]: `Hoek.assert(!err)` should be enough and more readable.

File: lib/index.js [hunk 8]
```diff
@@ -0,0 +1 @@
+// Stub file
```
Comment (hueniverse) [inline lib/index.js]: Should be the other way around. The root index.js should never have actual application code. Everything should be in lib (this makes working with IDEs easier when searching for code in the entire project). Either have the package.json file point directly to lib/index.js or use the hapi style of a root index.js that just loads lib/index.js.

File: package.json [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+  "main": "index.js",
+  "scripts": {
+    "test": "echo 'You should write some tests!'",
+    "start": "node index.js"
+  },
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/hueniverse/hueniversity.git"
+  },
+  "license": "ISC",
```
Comment (AdriVanHoudt) [inline package.json]: Shouldn't this be BSD?

**Pull request closed without merging**